sympy/matrices/matrices.py

```diff
@@ -1549,11 +1549,12 @@ def jordan_cell_power(jc, n):
             l = jc[0, 0]
             if (num < 0) == True and l == 0:
                 raise ValueError("Matrix det == 0; not invertible")
-            if (num < 2) == True:
-                b = self.jordan_cells()[1]
-                for i in range(len(b)):
-                    if (b[i]).rows > 1 and b[i][0, 0] == 0:
-                        raise ValueError("Cannot be evaluated")
+            if not isinstance(num, (int, Integer, Expr)):
+                if not num - int(num) == 0:
```

Comment:
Tried ```if not num.is_integer```, but throws Attribute error.
```
if not isinstance(num, (int, Integer)):
    if isinstance(num, float) and not num - int(num) == 0:
```
Is this better way to allow non negative integers and float which are integer exponents? Should ```Expr``` be allowed too?